### PR TITLE
feat: Create release process 

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

- Create release automation for busola

**Related issue(s)**

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- [x] Related issues are linked. To link internal trackers, use the issue IDs like `backlog#4567`
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging
https://github.com/kyma-project/busola/issues/3375

### CODE REVIEW

File: .github/workflows/busola-backend-build.yml
Comment (mrCherry97) [inline .github/workflows/busola-backend-build.yml:30]: make sure that we can remove this

File: .github/workflows/busola-web-build.yml
Comment (mrCherry97) [inline .github/workflows/busola-web-build.yml:45]: Make sure after first release on dev, that it is working and displayed correctly

File: .github/workflows/create-release.yml
Comment (mrCherry97) [inline .github/workflows/create-release.yml:36]: remove it for now, will be done in separate task

File: .github/workflows/create-release.yml
Comment (mrCherry97) [inline .github/workflows/create-release.yml:40]:
```suggestion
    needs: [ build-web, build-backend, build-local ]
```

File: Dockerfile.web
Comment (mrCherry97) [inline Dockerfile.web:19]: Add some comment to clarify what's going on

File: resources/apply-resources.sh
Comment (mrCherry97) [inline resources/apply-resources.sh:22]: Will be changed with https://github.com/kyma-project/busola/issues/3493

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
Check after merge if `latest` is build, and user can run Busola via docker using latest images.

Check internal docs `deploy-PR-on-cluster` if changes to kustomization will break our previous flow.

File: .github/workflows/create-release.yml
Comment (dbadura) [reply .github/workflows/create-release.yml:47]: Yes, according to docs:
```
# Number of commits to fetch. 0 indicates all history for all branches and tags.
    # Default: 1
    fetch-depth: ''
    ```
If we want to create changelog, it's needed

File: .github/workflows/busola-backend-build.yml
Comment (dbadura) [reply .github/workflows/busola-backend-build.yml]: `export-tags` is not needed in my opinion. According to docs:
```
description: Export parsed tags as build-args into dockerfile. Each tag will have format TAG_x, where x is the tag name passed along with the tag
```
I think that it's unnecessary complication and it hides the build logic.

**Review by mrCherry97 — COMMENTED** [review]
LGTM, waiting for second review to approve

**Review by mrCherry97 — CHANGES_REQUESTED** [review]
Please add documentation: https://github.com/kyma-project/busola/issues/3375#issuecomment-2499949784

**Comment by dbadura** [comment]
> Please add documentation: [#3375 (comment)](https://github.com/kyma-project/busola/issues/3375#issuecomment-2499949784)

I will create separate PR with docs and needed adjustment to busola and configs

**Review by mrCherry97 — APPROVED** [review]

**Pull request merged**
